File: java/v1/src/main/java/com/api/v1/book/dtos/NewBookRequestDto.java
package com.api.v1.book.dtos;

import com.api.v1.annotations.ISBN;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Min;

public record NewBookRequestDto(
        @NotBlank String title,
        String subtitle,
        @ISBN String isbn,
        @NotBlank String author,
        int publishingYear,
        @NotBlank String field,
        @Min(1) int numberOfPages,
        @Min(1) int version
) {
}

File: kotlin/v1/src/main/kotlin/com/api/v1/book/dtos/NewBookRequestDto.kt
package com.api.v1.book.dtos

import jakarta.validation.constraints.*

data class NewBookRequestDto(
    @NotBlank val title: String,
    val subtitle: String,
    @NotNull @Size(min=13, max=13) val isbn: String,
    @NotBlank val publisher: String,
    @NotEmpty val publishingYear: Int,
    @Min(1) val version: Int,
    @NotEmpty val numberOfPages: Int,
    @NotBlank val author: String,
    @NotBlank val field: String
)
